extensions/skyportal/static/js/components/VegaPlotZTFAlert.jsx

```diff
@@ -0,0 +1,244 @@
+        },
+        tooltip: [
+          // { field: "candid", title: "candid" },
+          { field: "magAndErr", title: "mag", type: "nominal" },
+          { field: "fid", type: "ordinal" },
+          { field: "jd", type: "quantitative" },
+          { field: "diffmaglim", type: "quantitative", format: ".2f" }
+        ],
+        opacity: {
+          condition: { selection: "filterMags", value: 1 },
+          value: 0
+        }
+      }
+    },
+
+    // Render error bars
+    {
+      selection: {
+        filterErrBars: {
+          type: "multi",
+          fields: ["fid"],
+          bind: "legend"
+        }
+      },
+      transform: [
+        { filter: "datum.magpsf != null && datum.sigmapsf != null" },
+        { calculate: "datum.magpsf - datum.sigmapsf", as: "magMin" },
+        { calculate: "datum.magpsf + datum.sigmapsf", as: "magMax" }
+      ],
+      mark: {
+        type: "rule",
+        size: 0.5
+      },
+      encoding: {
+        x: {
+          field: "jd",
+          type: "quantitative",
+          scale: {
+            zero: false,
+          }
+        },
+        y: {
+          field: "magMin",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true
+          }
+        },
+        y2: {
+          field: "magMax",
+          type: "quantitative",
+          scale: {
+            zero: false,
+            reverse: true
+          }
+        },
+        color: {
+          field: "fid",
+          type: "nominal"
+        },
+        opacity: {
+          condition: { selection: "filterErrBars", value: 1 },
+          value: 0
+        }
+      }
+    },
+
+    // Render limiting mags
+    {
+      transform: [
+        { filter: "datum.magpsf == null" }
+      ],
+      selection: {
+        filterLimitingMags: {
+          type: "multi",
+          fields: ["fid"],
+          bind: "legend"
+        }
+      },
+      mark: {
+        type: "point",
+        shape: "triangle-down"
+      },
+      encoding: {
+        x: {
+          field: "jd",
+          type: "quantitative",
+          scale: {
+            zero: false,
+          }
+        },
+        y: {
+          field: "diffmaglim",
+          type: "quantitative"
+        },
+        color: {
+          field: "fid",
+          type: "nominal"
+        },
+        tooltip: [
+          { field: "fid", type: "ordinal" },
+          { field: "jd", type: "quantitative" },
+          { field: "diffmaglim", type: "quantitative", format: ".2f" }
+        ],
+        opacity: {
+          condition: { selection: "filterLimitingMags", value: 0.3 },
+          value: 0
+        }
+      }
+    },
+
+    // render selected candid date
+    {
+      data: {values: [{}]},
+      mark: {type: "rule", strokeDash: [4, 4], size: 2},
+      encoding: {
+        x: {
+          datum: jd,
+          type: "quantitative"
+        }
+      }
+    }
+  ]
+});
+
+
+const VegaPlot = ({ dataUrl, jd }) => (
```

Comment:
Should probably have another name?

Reply:
In fact, can't we reuse VegaPlot from SkyPortal and just provide a different schema?

Reply:
Agreed

Reply:
Please teach me the ways to do that properly then, masters :) Specifically, I need an extra parameter passed to VegaPlot (jd) -- how do I reuse the one from SP?